.claude/hooks/stop-tests.sh

```
@@ -0,0 +1,30 @@
+# Exit 2 hands the failure back and Claude keeps working. The stop_hook_active
+# check prevents an infinite loop -- on the second pass it lets the turn end so
+# Claude can report the failure rather than grinding.
+set -u
+input=$(cat)
+active=$(echo "$input" | jq -r '.stop_hook_active // false')
+[ "$active" = "true" ] && exit 0   # already looped once; let it stop and report
+
+# Only run when something actually changed -- skips pure Q&A turns.
+# `git log @{u}..HEAD` errors when the branch has no upstream, so 2>/dev/null
+# makes a fresh branch read as "no unpushed commits" rather than aborting.
+if git diff --quiet && git diff --cached --quiet && [ -z "$(git log @{u}..HEAD 2>/dev/null)" ]; then
```

Shellcheck (0.11.0)

[warning] This { is literal. Check expression (missing ;/\n?) or quote it.

(SC1083)